### PR TITLE
results and app flow

### PR DESCRIPTION
**Pull request by AhsanBracu**
# Description
1.Result to category

### CODE REVIEW

**Comment by nvmanh** [comment]
So confused!
App flow:
1. In category activity: we have back button to return home screen
- click 1 category will start lesson of that category
2. Lesson activity
- Back button: return category activity (show confirm dialog that user want to cancel current lesson or not)
- Done button: finish current lesson activity to show result activity
3. Result Activity:
- We have NOT back button. User click back (android system) = Ok button to finish lesson and return to category activity

In here user start lesson from category:
1. with the same category (user not finish before) we start new lesson
2. with different category we start new lesson

Notice:
- You can not back any screen to Lesson activity
- If user are in result activity -> press OK button on top action bar or back button of android system) to return category activity -> if user press back again we will return home
- You can check this issue when you do many lesson then only press back button of android system you can many many activities which you start -> it's WRONG

You can repeat this flow many times without back to home screen (category -> lesson ->result ->category->lesson->result many times then press back button of android system)
Please remember flow:  category -> lesson->result
                                       category->lesson then back to category -> home -> exist (not show any activity such as last lesson or last result or last category activity)
                                       category press back -> home screen   (not show any activity such as last lesson or last result or last category activity)

Be careful. I will check you code directly.

File: app/src/main/java/com/framgia/elsytem/LearnedActivity.java [hunk 1]
```diff
@@ -129,6 +158,68 @@ public void onNothingSelected(AdapterView<?> parent) {
...
+                boolean t = newFolder.mkdir();
+                Log.e("file create ", " " + t);
+            }
+            //File pdfFolder = new File(Environment.getExternalStorageDirectory() + "/mydir/");
+            if (!pdfFolder.exists()) {
+                pdfFolder.mkdir();
+                //  pdfFolder.createFile();
+                Log.e("check", "Pdf Directory created" + sdCard.getAbsolutePath() + filepath);
+            }
+            myFile = new File(newFolder, "word_list.pdf");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LearnedActivity.java]: if file exist. You can name file = word_list + "_"+System.currentMilliseconds

File: app/src/main/java/com/framgia/elsytem/LearnedActivity.java [hunk 2]
```diff
@@ -129,6 +158,68 @@ public void onNothingSelected(AdapterView<?> parent) {
...
+                //document.newPage();
+            }
+            Chunk greenText = new Chunk(text, green);
+            Paragraph paragraph = new Paragraph(greenText);
+            paragraph.setAlignment(com.itextpdf.text.Element.ALIGN_CENTER);
+            document.add(paragraph);
+            document.close();
+        }
+    }
+    private static boolean isExternalStorageReadOnly() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LearnedActivity.java]: createpdf() isExternalStorageReadOnly() isExternalStorageAvailable() you can create Libs for these method. Don't write all methods in 1 activity class

Comment (AhsanBracu) [inline app/src/main/java/com/framgia/elsytem/LearnedActivity.java]: I don't understand the part of create Libs, can you explain it?

File: app/src/main/java/com/framgia/elsytem/mypackage/Methods.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.framgia.elsytem.mypackage;
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/mypackage/Methods.java]: rename this class to Libs or Utils please and avoid name as mypackage or yourname + classname....

**Comment by nvmanh** [comment]
Check flow in my description above. You do nothing to resolve this.
And another issue:
- Result activity: 
+ click Ok -> category activity (show dialog load category -> wrong)
+ click Back (android) -> do not show dialog load category -> true

**Comment by nvmanh** [comment]
Dialog Load category still show 2 times

**Pull request merged**